test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -1426,6 +1434,291 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
+      //   }
+      clientEncryption = new mongodbClientEncryption.ClientEncryption(keyVaultClient, {
+        keyVaultNamespace: 'keyvault.datakeys',
+        kmsProviders: getKmsProviders(LOCAL_KEY),
+        bson: BSON
+      });
+      // Create a MongoClient named ``encryptedClient`` with these ``AutoEncryptionOpts``:
+      //   AutoEncryptionOpts {
+      //     keyVaultNamespace: "keyvault.datakeys";
+      //     kmsProviders: { "local": { "key": <base64 decoding of LOCAL_MASTERKEY> } },
+      //     bypassQueryAnalysis: true
+      //   }
+      encryptedClient = this.configuration.newClient(
+        {},
+        {
+          autoEncryption: {
+            bypassQueryAnalysis: true,
+            keyVaultNamespace: 'keyvault.datakeys',
+            kmsProviders: getKmsProviders(LOCAL_KEY)
+          }
+        }
+      );
+    });
+
+    afterEach(async function () {
+      await setupClient.close();
+      await keyVaultClient.close();
+      await encryptedClient.close();
+    });
+
+    context('Case 1: can insert encrypted indexed and find', eeMetadata, function () {
+      let insertPayload;
+      let findPayload;
+
+      beforeEach(async function () {
+        // Use clientEncryption to encrypt the value "encrypted indexed value" with these EncryptOpts:
+        // class EncryptOpts {
+        //   keyId : <key1ID>
+        //   algorithm: "Indexed",
+        // }
+        // Store the result in insertPayload.
+        insertPayload = await clientEncryption.encrypt('encrypted indexed value', {
+          keyId: key1Id,
+          algorithm: 'Indexed'
+        });
+        // Use encryptedClient to insert the document { "encryptedIndexed": <insertPayload> }
+        // into db.explicit_encryption.
+        await encryptedClient.db('db').collection('explicit_encryption').insertOne({
+          encryptedIndexed: insertPayload
+        });
+        // Use clientEncryption to encrypt the value "encrypted indexed value" with these EncryptOpts:
+        // class EncryptOpts {
+        //    keyId : <key1ID>
+        //    algorithm: "Indexed",
+        //    queryType: Equality
+        // }
+        // Store the result in findPayload.
+        findPayload = await clientEncryption.encrypt('encrypted indexed value', {
+          keyId: key1Id,
+          algorithm: 'Indexed',
+          queryType: 'equality'
+        });
+      });
+
+      it('returns the decrypted value', async function () {
+        // Use encryptedClient to run a "find" operation on the db.explicit_encryption
+        // collection with the filter { "encryptedIndexed": <findPayload> }.
+        // Assert one document is returned containing the field
+        // { "encryptedIndexed": "encrypted indexed value" }.
+        const collection = encryptedClient.db('db').collection('explicit_encryption');
+        const result = await collection.findOne({ encryptedIndexed: findPayload });
+        expect(result.encryptedIndexed).to.equal('encrypted indexed value');
```

Comment:
this is an entirely optional suggestion, but since we are testing for equality, you could use the `expect(result).to.have.property('encryptedIndexed', 'encrypted indexed value')` for an assertion with a slightly better message in case of failure throughout this file

Reply:
Updated.